_scripts/update.py

```diff
@@ -55,8 +45,13 @@ def add_config(fx_root: str, fx_cfg_path: str, done: set[str]):
     return cfg_path
 
 
-def update_str(branch: str, fx_root: str, config_files: list[str]):
-    if branch not in [HEAD, "beta", "release"] and not match("esr[0-9]+", branch):
+def update_str(
+    branch: str,
+    fx_root: str,
+    cgf_automation: dict[str, list[str] | str],
```

Comment:
Is the type hint correct? I never use them, so it's kind of a guess 🤔

Reply:
It would be best to use a [TypedDict](https://docs.python.org/3/library/typing.html#typing.TypedDict) for this instead.

```python
from typing import TypedDict

class AutomationConfig(TypedDict):
    branches: list[str]
    head: str
    paths: list[str]
```

Also, in this context this would make more sense as the first rather then third argument.